packages/core/src/types.ts

```diff
@@ -198,6 +198,35 @@ export interface ActivityDefinition<TContext, TEvent extends EventObject>
 }
 
 export type Sender<TEvent extends EventObject> = (event: Event<TEvent>) => void;
+
+type ExcludeType<A> = { [K in Exclude<keyof A, 'type'>]: A[K] };
+
+type ExtractExtraParameters<A, T> = A extends { type: T }
+  ? ExcludeType<A>
+  : never;
+
+type ExtractSimple<A> = A extends any
+  ? {} extends ExcludeType<A>
+    ? A
+    : never
+  : never;
+
+type NeverIfEmpty<T> = {} extends T ? never : T;
+
+export interface PayloadSender<TEvent extends EventObject> {
```

Comment:
Keep in mind that **moving** this type into the core results in a problem for `@xstate/react` which starts to depend on this but its peer dependency range on `xstate` allows older XState versions that do not export this type.

The correct way to do this would be to:
- keep the type duplicated in both packages for now
- or create the `@xstate/types` dependency that would be specified as a regular dependency for both packages and as the result the problem would go away once and for all